lib/repository.php

```diff
@@ -55,9 +55,11 @@ public static function add($entity, $save = true) {
 
     public static function edit($id, $entity) {
         if (array_key_exists($id, self::$entities)) {
-            self::$entities[$id] = array_merge(self::$entities[$id], $entity);
-            self::save();
-            return true;
+            if (is_array(self::$entities[$id]) && is_array($entity)) {
```

Comment:
If these are objects (e.g. Hm_Contacts), I think you should replace the object in the entities list - otherwise, the change would be discarded and not saved.

Reply:
It seems that I did not understand correctly

Reply:
Why do you add the is_array check? Most probably because the entity might not be an array but Hm_Content (object) in which case you should replace the object in the list. The purpose of this edit method is to update an entity in the list. If you ignore the non-array ones, you fail to deliver the purpose of the method and silently introduce bugs.